### ISSUE TITLE
Training Collapse

### ISSUE THREAD

**Issue by Ching-Yee-Chan** [facebookresearch/audioseal#55]
I am trying to reproduce the training process from scratch on Voxpopuli-en. I preserved all your original hyperparameters but found that watermark-related losses stayed the same even after 30 epoches. Conversely, the g_loss went down quickly to negative.
![Screenshot 2024-10-14 094216](https://github.com/user-attachments/assets/d56538a4-08fe-4241-b675-d40f7cd86fb8)
I noticed that since the watermark is added directly on the original waveform, there is a shortcut from the original audio to the watermarked audio. Therefore, the model will try to ignore the watermark and overfit g_loss if the gradient of watermark-related losses is considerably small. So I tried to:

1. Set weight of wm_detection and wm_mb to 200 and tf_loudnessratio to 2000 (200×). Watermark loss went down in 10 epoches whereas the adversarial loss went up afterwards, introducing significant noise.
2. Turning off the balancer (which significantly scales loss≈0 up). No difference.
3. Rescale the gradient of g_loss so that r=norm(g_loss)/norm(wm_loss) is fixed as a hyperparameter. However I found that this makes no difference until r<0.1 where  g_loss exploded.

Settings:

I ran `dora run solver=watermark/robustness dset=audio/voxpopuli` on a single 48GB GPU. Since I do not have access to any Slurm clusters, running dora hyperparameter search may not be feasible. All hyperparameters follows `config/solver/watermark/default.yaml` except those mentioned above.

Any of your insights or suggestions on this problem would be appreciated.

**Comment by pierrefdz** [comment]
Hi!
Sorry for the delay in answering. Have you found anything?

The training was quite unstable, even in our case, so maybe the best hyperparams are not the ones that we put depending on your GPU configurations and batchsizes. I would suggest to try different LR and optimizers, and maybe even different seeds to see if the problem has a deeper root than just unluckiness.
